src/Distributed/partition_assemble.jl

```diff
@@ -110,10 +110,11 @@ end
 Partition a global array in local arrays of size `(nx, ny)` if 2D or `(nx, ny, nz)` is 3D.
 Usefull for boundary arrays, forcings and initial conditions.
 """
-partition_global_array(arch, c_global::Function, n) = c_global 
+partition_global_array(arch, c_global::AbstractArray, n) = c_global
+partition_global_array(arch, c_global::Function, n)      = c_global 
 
 # Here we assume that we cannot partition in z (we should remove support for that)
-function partition_global_array(arch, c_global::AbstractArray, n) 
+function partition_global_array(arch::DistributedArch, c_global::AbstractArray, n) 
```

Comment:
should we call this `DistributedArchitecture`? Wouldn't want to confuse it with

<img width="246" alt="image" src="https://github.com/CliMA/Oceananigans.jl/assets/15271942/18aec7f5-24a0-4115-b78d-c8e70e420c2b">

which can cause trouble with Italians.

Reply:
what about `MultiProcessor`?

Reply:
A minor improvement on that would be `MultiProcess`. (When you say "multi-processor", I am thinking of the device itself, rather than an abstraction that represents the whole computation)